pxd.c

```diff
@@ -1778,9 +1798,17 @@ static ssize_t pxd_remove_dev(struct fuse_conn *fc, uint64_t dev_id, bool force)
 	}
 
 	if (!pxd_dev->removing) {
+		if (atomic_cmpxchg(&pxd_dev->fp.ioswitch_active, 1, 0) != 0) {
+			err = -EBUSY;
+			goto out_lock;
+		}
+		printk(KERN_DEBUG "device %llu detach: acquired ioswitch_active\n", pxd_dev->dev_id);
 		pxd_dev->removing = true;
 		INIT_WORK(&pxd_dev->remove_work, pxd_finish_remove);
 		schedule_work(&pxd_dev->remove_work);
+		// Release ioswitch lock
+		printk(KERN_DEBUG "device %llu detach: releasing ioswitch_active\n", pxd_dev->dev_id);
+		atomic_set(&pxd_dev->fp.ioswitch_active, 0);
```

Comment:
the above is unnecessary - remove it. ioswitch_active always remains active when device is being detached.